src/UI/Component/Chart/Speedo/Speedo.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/* Copyright (c) 2017 Ralph Dittrich <dittrich@qualitus.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Chart\Speedo;
+use ILIAS\UI\Component\Component;
+
+/**
+ * Interface Speedo
+ * @package ILIAS\UI\Component\Chart\Speedo
+ */
+interface Speedo extends Component {
+
+    /**
+     * Get maximum value
+     *
+     * This value is used as 100%.
+     * This value will always returned "raw" because it is used to calculate the
+     * percentage values of score, minimum and diagnostic.
+     *
+     * @return int|float
+     */
+    public function getMaximum();
+
+    /**
+     * Get minimum value
+     *
+     * This value represents the minimum score that users need, to fulfill the objective.
+     * If this value is not set, it defaults to 100%.
+     *
+     * @param bool $getAsPercent Get Value as percentage value or not
+     * @return int|float
+     */
+    public function getMinimum($getAsPercent = true);
+
+    /**
+     * Get user score value
+     *
+     * This value represents the users score. It is rendered as the speedo main bar.
+     *
+     * @param bool $getAsPercent Get Value as percentage value or not
+     * @return int|float
+     */
+    public function getScore($getAsPercent = true);

```

Comment:
I used to make the switch between percent and "raw" value as variable, because I wanted a simple way to get the one or the other value without having to call multiple functions. So you may have three functions, that get true or false, instead of six functions and an extra if clause.
This seemed easier to use to me. But if you think to split it up to multiple functions, would be the better way, I do not mind to change it.